### PR TITLE
Fixing fix_test/test_target_map_pointer.c to use headerfile

### PR DESCRIPTION
**Pull request by tmh97**
Fixing test_target_map_pointer.c to use headerfile, as per issue #10.

### CODE REVIEW

**Comment by josemonsalve2** [comment]
I suggest we improve this test by adding some additional checks that the pointer and array are the same. 

One option involves using a target data region which I am not sure it is a good idea to combine, but I feel it will add an extra element to the tests:


```C
int array[N], *p;
p = array;
#pragma omp target data map(tofrom:array)
#pragma omp target map(alloc: p[0:N])
for (...) 
   p[i] ++;

#pragma omp target map(alloc: array)
for (...)
   p[i] ++; 
```

The second option involves having the pointer being a function parameter

```C

void some_func(int * p) {
#pragma omp target map(alloc: p[0:N])
for (...) 
   p[i] ++;
}
main () {
  int array[N];
  p = array;
  #pragma omp target data map(tofrom:array)
   some_func(array);

  #pragma omp target map(alloc: array)
  for (...)
     p[i] ++; 
...
}
```

We should still keep the current target region that just uses the pointer.

**Comment by josemonsalve2** [comment]
Just to consider, the test I suggested might be a `target data` test instead of a `target` test, so it could even be a second PR. Up to you

File: tests/4.5/target/test_target_map_pointer.c [hunk 1]
```diff
@@ -1,36 +1,40 @@
...
 #pragma omp target data map(tofrom: compute_array) //To test default pointer behavior, array must be mapped before the pointer
-#pragma omp target map(to: p[:N]) map(tofrom: isHost)
+#pragma omp target map(to: p[:N]) 
   {
-    // Record where the computation was executed
-    isHost = omp_is_initial_device();
-  
     // Array modified through the pointer
     for (i = 0; i < N; i++)
       p[i] = i;
```
Comment (josemonsalve2) [inline tests/4.5/target/test_target_map_pointer.c]: See my ticket comment, and figure if it is better to do those changes in `test/target_data` adding a new test, or if you want to modify this one. I think that creating the new test might be a better idea.

Comment (tmh97) [reply tests/4.5/target/test_target_map_pointer.c]: I think to create a separate test is the best way to proceed.

**Comment by josemonsalve2** [comment]
Do you want to add it to the same PR?

**Comment by tmh97** [comment]
Would you recommend I create a new PR since it is an entirely new test and not a fix like the previous, or keep them together since they will be very similar?

**Comment by josemonsalve2** [comment]
I recommend keeping them together

**Comment by tmh97** [comment]
@jhdavis8 @josemonsalve2 Test passes fine with gcc, but when compiling with clang the tests fails on device with large error value (sum != result). Have either of you seen issues with how clang handles pointers, or any issue with the code that would cause to fail on clang?

**Comment by josemonsalve2** [comment]
We should discuss this further. I feel like this should work, but I am also confused if there is a need to use "`use_device_pointer()`" in the `target data` region

File: tests/4.5/target/test_target_map_pointer.c [hunk 2]
```diff
@@ -1,36 +1,40 @@
...
 #include <omp.h>
+#include "ompvv.h"
 
 #define N 1000
 
 int main() {
   int compute_array[N], *p;
   int sum = 0, result = 0;
-  int i, isHost = -1;
+  int i, errors;
```
Comment (spophale) [inline tests/4.5/target/test_target_map_pointer.c]: Initialize errors to 0

Comment (tmh97) [reply tests/4.5/target/test_target_map_pointer.c]: Thank you Swaroop!

**Comment by tmh97** [comment]
New changes tested on Fatnode, tested and passed with gcc and clang

**Pull request merged**